src/core-services/account/mutations/addAccountEmailRecord.js

```diff
@@ -32,8 +32,12 @@ export default async function addAccountEmailRecord(context, input) {
   const user = await users.findOne({ _id: account.userId });
   if (!user) throw new ReactionError("not-found", "User not Found");
 
-  if (!context.isInternalCall && userIdFromContext !== account.userId) {
-    await checkPermissions(["reaction-accounts"], account.shopId);
+  if (!context.isInternalCall) {
+    await context.validatePermissions(`reaction:accounts:${account._id}`, "add:emails", {
+      shopId: account.shopId,
+      owner: account._id,
```

Comment:
@rosshadden @kieckhafer All `validatePermissions` calls should be using `account.userId` rather than `account._id` everywhere. We currently set them to the same string, but it's not a guarantee if people import them or something.

Reply:
Understood, thank you!

Reply:
Referring to `owner` value. In this case `reaction:accounts:${account._id}` probably is correct because that's the account you are accessing.

Reply:
What about `context.accountId`? We have a lot of properties that are (currently) the same thing.